sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/query.sql

```diff
@@ -0,0 +1,131 @@
+--STEP 1: Get clients with a first seen date = submission date - 14 days
+--Note: Min cohort date is 2023-11-01 so backfilling will return nothing before then
+--Note: Max cohort date cannot be more than 7 days ago (to ensure we always have at least 7 days of data)
+WITH clients_first_seen_14_days_ago AS (
+  SELECT
+    client_id,
+    first_seen_date,
+    country,
+    attribution_campaign,
+    attribution_content,
+    attribution_dltoken,
+    attribution_medium,
+    attribution_source
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_first_seen` --contains all new clients, including those that never sent a main ping
+  WHERE
+    first_seen_date = @submission_date --this is 14 days before {{ds}}
+    AND first_seen_date
+    BETWEEN '2023-11-01'
+    AND DATE_SUB(CURRENT_DATE, INTERVAL 8 DAY)
+),
+--Step 2: Get only the columns we need from clients last seen, for only the small window of time we need
+clients_last_seen_raw AS (
+  SELECT
+    cls.client_id,
+    cls.first_seen_date,
+    cls.country,
+    cls.submission_date,
+    cls.days_since_seen,
+    cls.active_hours_sum,
+    cls.days_visited_1_uri_bits,
+    cls.days_interacted_bits,
+    cls.search_with_ads_count_all
+  FROM
```

Comment:
Should this be a left join, since we want to retain the clients from `clients_first_seen_14_days_ago` that didn't send a main ping? I think this will `INNER JOIN` and thus drop clients without an entry in `cls`?

```sql
FROM 
     clients_first_seen_14_days_ago clients
LEFT JOIN    
     `moz-fx-data-shared-prod.telemetry.clients_last_seen` cls
ON clients.client_id = cls.client_id 
```

Reply:
Oh I see now, you are doing this to limit the result from CLS, given the LEFT JOIN on line 109.